Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/core/TestCertificateProcessor.kt

```diff
@@ -51,30 +49,31 @@ class TestCertificateProcessor @Inject constructor(
             return data
         }
 
-        val rsaKeyPair = try {
-            rsaKeyPairGenerator.generate()
-        } catch (e: Throwable) {
-            throw InvalidTestCertificateException(InvalidHealthCertificateException.ErrorCode.RSA_KP_GENERATION_FAILED)
-        }
+        val publicKey = data.rsaPublicKey ?: throw IllegalArgumentException("rsaPublicKey was null")
```

Comment:
How is this resolved to a human readable error?

Reply:
It's not resolved, but it is also not something that should be able to happen.

Every test with `publicKeyRegisteredAt == null` came from `TestCertificateRepository.requestCertificate` and has a keypair:

https://github.com/corona-warn-app/cwa-app-android/pull/3650/files/82e7e7bdb87457048027636a02e7472bdf47d668#diff-91615199099cc98a19cabc6f30424cabf97e9663712eb96a68d38b25efba977fR140-R142

The keypair can only be null, if the certificate was added via qrcode and then  `publicKeyRegisteredAt` is not null, 

https://github.com/corona-warn-app/cwa-app-android/pull/3650/files/82e7e7bdb87457048027636a02e7472bdf47d668#diff-91615199099cc98a19cabc6f30424cabf97e9663712eb96a68d38b25efba977fR183

and we should not get here due to:
https://github.com/corona-warn-app/cwa-app-android/pull/3650/files/82e7e7bdb87457048027636a02e7472bdf47d668#diff-3b73348d0d181c4b360ab8489616832b6f4e8f1dd5936f7f25eaa585e19ab048R47-R50

Reply:
You are right, already affected users would see the "not something that should be able to happen" error :thinking: